packages/garbo/src/resources/mayam.ts

```diff
@@ -0,0 +1,262 @@
+    ring: 1,
+  },
+  {
+    choice: "chair",
+    value: () => 0, // +5 free rests, I don't know how to value that?
+    ring: 1,
+  },
+  {
+    choice: "fur",
+    value: () =>
+      Math.max(
+        0,
+        ...getExperienceFamiliars("free").map(
+          ({ expectedValue }) => expectedValue / 12,
+        ),
+      ) * 100,
+    ring: 1,
+  },
+  {
+    choice: "vessel",
+    value: () => 0, // How do we value MP regen?
+    ring: 1,
+  },
+  {
+    choice: "lightning",
+    value: () => 0,
+    ring: 2,
+  },
+  {
+    choice: "bottle",
+    value: () => 0, // How do we value HP regen?
+    ring: 2,
+  },
+  {
+    choice: "meat",
+    value: () => clamp(myLevel() * 100, 100, 1500),
+    ring: 2,
+  },
+  {
+    choice: "wood",
+    value: () => 0,
+    ring: 2,
+  },
+  {
+    choice: "wall",
+    value: () => 0,
+    ring: 3,
+  },
+  {
+    choice: "cheese",
+    value: () => garboValue($item`goat cheese`),
+    ring: 3,
+  },
+  {
+    choice: "eyepatch",
+    value: () => 0,
+    ring: 3,
+  },
+  {
+    choice: "explosion",
+    value: () => 0,
+    ring: 4,
+  },
+  {
+    choice: "clock",
+    value: () => {
+      return (globalOptions.prefs.valueOfAdventure ?? 4000) * 5;
+    },
+    ring: 4,
+  },
+];
+
+// Check whether any individual ring is used before adding it to the list for MaxBy
+const AVAILABLE_RING_OPTIONS: MayamRingSummon[] = MAYAM_RING_OPTIONS.filter(
+  (option) => {
+    return !yamUses().includes(option.choice);
+  },
+);
+
+// Check whether any component of a resonance is used before adding it to the list for MaxBy
+function isResonanceAvailable(resonance: Resonance): boolean {
+  const usedChoices = yamUses()
+    .split(",")
+    .map((choice) => choice.trim());
+  const resonanceChoices = resonance.rings.split(" ");
+  return resonanceChoices.every((choice) => !usedChoices.includes(choice));
+}
+
+// Filter resonances to only include available ones
+const AVAILABLE_RESONANCES = () => RESONANCE.filter(isResonanceAvailable);
+
+function findBestRingValue():
+  | MayamCalendar.CombinationString
+  | MayamCalendar.Combination {
+  const filteredOptions = AVAILABLE_RING_OPTIONS;
+
+  // Check for the highest value in each ring, filtering by ring
+  const ring1 = maxBy(
+    filteredOptions.filter((option) => option.ring === 1),
+    (option) => option.value(),
+  );
+  const ring2 = maxBy(
+    filteredOptions.filter((option) => option.ring === 2),
+    (option) => option.value(),
+  );
+  const ring3 = maxBy(
+    filteredOptions.filter((option) => option.ring === 3),
+    (option) => option.value(),
+  );
+  const ring4 = maxBy(
+    filteredOptions.filter((option) => option.ring === 4),
+    (option) => option.value(),
+  );
+
+  // Check for the highest value resonance
+  const maxResonance = maxBy(AVAILABLE_RESONANCES(), (resonance) =>
+    findResonanceValue(resonance.rings),
+  );
+
+  // Sum the total ring value and determine the resonance value
+  const totalRingValue =
+    (ring1?.value() || 0) +
```

Comment:
```suggestion
    (ring1?.value() ?? 0) +
```

On one hand, you should use nullish coalescence. On the other hand, what you really should do is use `sumNumbers` and define all four of your rings at once by saying `const rings = new Array(4).fill().map((_, i) => maxBy(filteredOptions.filter((option) => option.ring === i + 1), (option) => option.value());`. On the third hand, it shouldn't even look exactly like that, because I think having `filteredOptions` be one giant hodgepodge array feels wrong.